storage/s3/src/main/java/io/aiven/kafka/tieredstorage/storage/s3/MetricCollector.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2023 Aiven Oy
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.tieredstorage.storage.s3;
+
+import java.util.List;
+
+import org.apache.kafka.common.metrics.JmxReporter;
+import org.apache.kafka.common.metrics.KafkaMetricsContext;
+import org.apache.kafka.common.metrics.MetricConfig;
+import org.apache.kafka.common.metrics.Sensor;
+import org.apache.kafka.common.metrics.stats.CumulativeCount;
+import org.apache.kafka.common.metrics.stats.Rate;
+import org.apache.kafka.common.utils.Time;
+
+import com.amazonaws.Request;
+import com.amazonaws.Response;
+import com.amazonaws.metrics.RequestMetricCollector;
+import com.amazonaws.services.s3.model.AbortMultipartUploadRequest;
+import com.amazonaws.services.s3.model.CompleteMultipartUploadRequest;
+import com.amazonaws.services.s3.model.DeleteObjectRequest;
+import com.amazonaws.services.s3.model.GetObjectRequest;
+import com.amazonaws.services.s3.model.InitiateMultipartUploadRequest;
+import com.amazonaws.services.s3.model.PutObjectRequest;
+import com.amazonaws.services.s3.model.UploadPartRequest;
+import com.amazonaws.util.AWSRequestMetrics;
+
+class MetricCollector extends RequestMetricCollector {
+    private final org.apache.kafka.common.metrics.Metrics metrics;
+
+    private final Sensor getObjectRequests;
+    private final Sensor putObjectRequests;
+    private final Sensor deleteObjectRequests;
+
+    private final Sensor createMultipartUploadRequests;
+    private final Sensor uploadPartRequests;
+    private final Sensor completeMultipartUploadRequests;
+    private final Sensor abortMultipartUploadRequests;
+
+    MetricCollector() {
+        final JmxReporter reporter = new JmxReporter();
+
+        metrics = new org.apache.kafka.common.metrics.Metrics(
+            new MetricConfig(), List.of(reporter), Time.SYSTEM,
+            new KafkaMetricsContext("aiven.kafka.server.tieredstorage.s3")
+        );
+        final String metricGroup = "s3-metrics";
```

Comment:
This can e constant I guess and we could get rid of additional argument in `createSensor`?

Reply:
extracted